app/controllers/users_controller.rb

```diff
@@ -59,7 +57,11 @@ def destroy
 
   private
 
-  def user_params
+  def set_user
+    @user = User.find(params[:id])
+  end
+
+  def allowed_user_params
     if current_user.admin?
```

Comment:
I'd probably consider using the conditional check to merge in a value, like below:

```
def allowed_user_params
  allowed_fields = %i( name email password password_confirmation tone_name band_start_date )
  allowed_fields << :admin if current_user.admin?
  params.require(:user).permit(*allowed_fields)
end
```